prometheus/counter.go

```diff
@@ -207,6 +213,7 @@ func (v2) NewCounterVec(opts CounterVecOpts) *CounterVec {
 			}
 			result := &counter{desc: desc, labelPairs: MakeLabelPairs(desc, lvs), now: time.Now}
```

Comment:
Something is odd here - I expect `now`: being passed from CounterVecOpts, so you can test this code.

Reply:
the `now` function from `counter` is not used to initialize the created timestamp. It is used only for exemplars.

The `now` function from `CounterVecOpts` in the other hand is used further below

Reply:
This is still quite wrong. Do we need multiple now functions?

Reply:
I believe we do, since `counter` doesn't have access to the `counterOpts` object that created it.

[Counter still needs to call `now()` when updating exemplars](https://github.com/prometheus/client_golang/blob/51d24f86807c24ce866355d0c82258420f571c17/prometheus/counter.go#L170)